src/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.BIO.Custom.cs

```diff
@@ -0,0 +1,163 @@
+        {
+            _createDelegate = Create;
+            _controlDelegate = Control;
+            _destroyDelegate = Destroy;
+            _writeDelegate = Write;
+            _readDelegate = Read;
+
+            var name = Marshal.StringToHGlobalAnsi("Managed Bio");
+            var bioStruct = new bio_method_st()
+            {
+                create = _createDelegate,
+                name = name,
+                type = BIO_TYPE.BIO_TYPE_SOURCE_SINK,
+                destroy = _destroyDelegate,
+                ctrl = _controlDelegate,
+                bread = _readDelegate,
+                bwrite = _writeDelegate,
+            };
+
+            var memory = Marshal.AllocHGlobal(Marshal.SizeOf<bio_method_st>());
+            Marshal.StructureToPtr(bioStruct, memory, true);
+            _customBioMethodStructure = memory;
+        }
+
+        internal static GCHandle BioGetGCHandle(IntPtr bio)
+        {
+            var ptr = Crypto.BioGetAppData(bio);
+            if (ptr == IntPtr.Zero)
+            {
+                return default(GCHandle);
+            }
+            var handle = GCHandle.FromIntPtr(ptr);
+            return handle;
+        }
+
+        internal static void BioSetGCHandle(SafeBioHandle bio, GCHandle handle)
+        {
+            var ptr = GCHandle.ToIntPtr(handle);
+            Crypto.BioSetAppData(bio, ptr);
+        }
+
+        internal static SafeBioHandle CreateCustomBio()
+        {
+            var bio = Crypto.CreateCustomBio(_customBioMethodStructure);
+            return bio;
+        }
+
+        private unsafe static int Create(bio_st* bio)
+        {
+            bio[0].init = 1;
+            return 1;
+        }
+
+        private static int Destroy(IntPtr bio) => 1;
+
+        private static unsafe long Control(IntPtr bio, Crypto.BIO_CTRL cmd, long param, void* ptr)
+        {
+            switch (cmd)
+            {
+                case Crypto.BIO_CTRL.BIO_CTRL_FLUSH:
+                case Crypto.BIO_CTRL.BIO_CTRL_POP:
+                case Crypto.BIO_CTRL.BIO_CTRL_PUSH:
+                    return 1;
+            }
+            return 0;
+        }
+
+        private static unsafe int Write(IntPtr bio, void* input, int size)
+        {
+            var handle = BioGetGCHandle(bio);
+
+            if (handle.IsAllocated && handle.Target is SafeSslHandle.WriteBioBuffer buffer)
+            {
+                return buffer.Write(new Span<byte>(input, size));
+            }
+
+            return -1;
+        }
+
+        private static unsafe int Read(IntPtr bio, void* output, int size)
+        {
+            var handle = BioGetGCHandle(bio);
+            if (handle.IsAllocated && handle.Target is SafeSslHandle.ReadBioBuffer buffer)
+            {
+                return buffer.Read(new Span<byte>(output, size));
+            }
+            return -1;
+        }
+
+        [Flags]
+        private enum BIO_TYPE
+        {
+            BIO_TYPE_SOURCE_SINK = 0x0400,
```

Comment:
I have removed all the enums as they are no longer needed with the new shim methods.